config/management/operational/src/network_checker.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_management::error::Error;
+use libra_network_address::NetworkAddress;
+use std::{
+    io::{Read, Write},
+    net::{TcpStream, ToSocketAddrs},
+    time::Duration,
+};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct CheckEndpoint {
+    #[structopt(long)]
+    address: NetworkAddress,
+}
+
+// This will show up as an error in the logs as a bad key 07070707...
+// It allows us to ensure that both connections work, and that we can see them in the logs
+const INVALID_NOISE_HEADER: &[u8; 152] = &[7; 152];
+
+impl CheckEndpoint {
+    pub fn execute(self) -> Result<String, Error> {
+        let addrs = self.address.to_socket_addrs().map_err(|err| {
+            Error::IO(
+                "Failed to resolve address from NetworkAddress".to_string(),
+                err,
+            )
+        })?;
+
+        let mut last_error = std::io::Error::new(std::io::ErrorKind::Other, "");
```

Comment:
didn't want to use an unwrap at the end :O

Reply:
can easily put in an unwrap_or_else(|| ... no addresses provided or some other nonsense)

Reply:
Yeah, I guess i could have done that :)